core/src/price.ts

```diff
@@ -49,3 +49,39 @@ export const calculateTransactionProfit = function (auction: Auction): BigNumber
     const totalMarketPriceLimitedByDebt = collateralAmountLimitedByDebt.multipliedBy(auction.marketUnitPrice);
     return totalMarketPriceLimitedByDebt.minus(auction.debtDAI);
 };
+
+export const calculateTransactionProfitDate = function (auction: Auction): Date | undefined {
+    if (
+        auction.secondsBetweenPriceDrops === undefined ||
+        auction.secondsTillNextPriceDrop === undefined ||
+        auction.marketUnitPrice === undefined ||
+        auction.priceDropRatio === undefined
+    ) {
+        return undefined;
+    }
+
+    const isAlreadyProfitable = auction.approximateUnitPrice.isLessThan(auction.marketUnitPrice);
+    const date = new Date();
+
+    let steps = 0;
+    let currentValue = auction.approximateUnitPrice;
+    let secondsTillProfitable = 0;
+
+    if (isAlreadyProfitable) {
+        while (currentValue.isLessThan(auction.marketUnitPrice)) {
+            steps -= 1;
+            currentValue = currentValue.multipliedBy(new BigNumber(1).minus(auction.priceDropRatio).plus(1));
+        }
+        secondsTillProfitable = auction.secondsBetweenPriceDrops * steps + auction.secondsTillNextPriceDrop;
+    } else {
+        while (currentValue.isGreaterThan(auction.marketUnitPrice)) {
+            steps += 1;
+            currentValue = currentValue.multipliedBy(auction.priceDropRatio);
+        }
+        const secondsSinceLastPriceDrop = auction.secondsBetweenPriceDrops - auction.secondsTillNextPriceDrop;
+        secondsTillProfitable = auction.secondsBetweenPriceDrops * steps - secondsSinceLastPriceDrop;
+    }
+
+    date.setSeconds(date.getSeconds() + secondsTillProfitable);
+    return date;
```

Comment:
- please rename `date` to something less generic or just avoid creating this extra variable
- please don't use `setSeconds` since it's actually should accept values [only between 0 and 59](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setSeconds#parameters)
- please use `addSeconds` instead https://date-fns.org/v2.28.0/docs/addSeconds